<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2019-09-05" />

<title>DS Analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">LPS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">DS Analysis</h1>
<h4 class="author">Helena L. Crowell</h4>
<address class="author_afil">
Institute for Molecular Life Sciences, University of Zurich, Switzerland<br><h4 class="author">Pierre-Luc Germain</h4>
<address class="author_afil">
Institute for Molecular Life Sciences, University of Zurich, Switzerland<br><h4 class="date">September 05, 2019</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-09-05
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>LPS/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190909code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190909)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190909code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190909)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
ds
</li>
<li>
load-data
</li>
<li>
load-libs
</li>
<li>
pb
</li>
<li>
save
</li>
<li>
session-info-chunk-inserted-by-workflowr
</li>
<li>
unnamed-chunk-2
</li>
<li>
unnamed-chunk-3
</li>
<li>
unnamed-chunk-4
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>3-differential_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomHelenaLCBC22019workshopmultiscRNAseqtreec93b9fb17c7a3c3b7aa9a3b2bc8d4c2cfd2288b8targetblankc93b9fba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/HelenaLC/BC2_2019-workshop_multi-scRNA-seq/tree/c93b9fb17c7a3c3b7aa9a3b2bc8d4c2cfd2288b8" target="_blank">c93b9fb</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomHelenaLCBC22019workshopmultiscRNAseqtreec93b9fb17c7a3c3b7aa9a3b2bc8d4c2cfd2288b8targetblankc93b9fba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    cheatsheet.log

Untracked files:
    Untracked:  .gitignore
    Untracked:  LPS/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level1">
<h1>Load packages</h1>
<pre class="r"><code>library(dplyr)
library(muscat)
library(scater)
library(SingleCellExperiment)
library(UpSetR)</code></pre>
</div>
<div id="load-data" class="section level1">
<h1>Load data</h1>
<pre class="r"><code>sce &lt;- readRDS(file.path(&quot;output&quot;, &quot;3-SCE_clustered.rds&quot;))

# make WT reference group
sce$group_id &lt;- relevel(sce$group_id, ref = &quot;WT&quot;)

# reorder sample levels
m &lt;- match(levels(sce$sample_id), sce$sample_id)
o &lt;- order(sce$group_id[m])
sce$sample_id &lt;- factor(sce$sample_id, levels = levels(sce$sample_id)[o])

# prep. SCE for &#39;muscat&#39;
sce &lt;- prepSCE(sce,
  cluster_id = &quot;cluster_id&quot;,
  sample_id = &quot;sample_id&quot;,
  group_id = &quot;group_id&quot;,
  drop = FALSE)</code></pre>
</div>
<div id="aggregation" class="section level1">
<h1>Aggregation</h1>
<pre class="r"><code># compute pseudobulk (sum of counts) by cluster-sample
(pb &lt;- aggregateData(sce, assay = &quot;counts&quot;, 
    by = c(&quot;cluster_id&quot;, &quot;sample_id&quot;), fun = &quot;sum&quot;))</code></pre>
<pre><code>class: SingleCellExperiment 
dim: 27998 8 
metadata(3): experiment_info agg_pars n_cells
assays(7): Astrocytes Endothelial ... Neuronal_excit
  Neuronal_inhib
rownames(27998): ENSMUSG00000051951.Xkr4 ENSMUSG00000089699.Gm1992
  ... ENSMUSG00000096730.Vmn2r122
  ENSMUSG00000095742.CAAA01147332.1
rowData names(0):
colnames(8): LC016_WT LC019_WT ... LC023_LPS LC026_LPS
colData names(2): group_id orig.ident
reducedDimNames(0):
spikeNames(0):
altExpNames(0):</code></pre>
<pre class="r"><code># pseudobulk-level MDS plot
pbMDS(pb)</code></pre>
<p><img src="figure/3-differential.Rmd/pb-1.png" width="576" style="display: block; margin: auto;" /></p>
</div>
<div id="ds-analysis" class="section level1">
<h1>DS analysis</h1>
<pre class="r"><code># run edgeR on pseudobulks
res &lt;- pbDS(pb, method = &quot;edgeR&quot;, verbose = FALSE)

# reformat results
tbl &lt;- resDS(sce, res, bind = &quot;col&quot;)

# filter &amp; sort
tbl &lt;- filter(tbl, p_adj.loc &lt; 0.05, logFC &gt; 1)
tbl &lt;- arrange(tbl, p_adj.loc)

# no. of DS genes per cluster
res_by_k &lt;- split(tbl, tbl$cluster_id)
vapply(res_by_k, nrow, numeric(1))</code></pre>
<pre><code>      Astrocytes      Endothelial        Microglia   Neuronal_excit 
             523              555              307               94 
  Neuronal_inhib Oligodendrocytes              OPC 
              62              192              153 </code></pre>
<pre class="r"><code># top hits in ea. cluster
top &lt;- do.call(&quot;rbind&quot;, lapply(res_by_k, head, 3))
top &lt;- select(top, -c(&quot;contrast&quot;, &quot;p_adj.glb&quot;, &quot;F&quot;))
top$gene &lt;- gsub(&quot;^.*\\.&quot;, &quot;&quot;, top$gene)
format(data.frame(top, row.names = NULL), digits = 3)</code></pre>
<pre><code>     gene       cluster_id logFC logCPM    p_val p_adj.loc
1  Cxcl10       Astrocytes 10.40   8.44 2.99e-61  5.43e-57
2    Osmr       Astrocytes  7.45   7.18 2.53e-41  2.30e-37
3   Sbno2       Astrocytes  5.03   8.14 5.89e-41  3.56e-37
4    Ccl2      Endothelial 10.80  10.10 2.52e-43  4.03e-39
5    Csf3      Endothelial 10.50   9.86 3.37e-42  2.69e-38
6   Cxcl1      Endothelial  9.19  10.20 1.93e-41  1.03e-37
7    Ccl5        Microglia 10.40   9.77 9.76e-25  1.63e-20
8  Cxcl10        Microglia  9.56   8.93 1.05e-23  8.72e-20
9   Il12b        Microglia  9.43   8.82 2.13e-23  1.18e-19
10   Lcn2   Neuronal_excit  7.54   4.24 4.82e-10  3.38e-06
11    Mt2   Neuronal_excit  2.32   7.35 4.82e-10  3.38e-06
12 Cxcl10   Neuronal_excit  5.57   3.15 7.77e-10  4.09e-06
13  Stat3   Neuronal_inhib  2.03   6.67 4.48e-15  8.96e-11
14    Mt2   Neuronal_inhib  2.26   7.80 4.73e-14  4.73e-10
15   Lcn2   Neuronal_inhib  5.83   3.91 1.69e-12  1.12e-08
16   Aff1 Oligodendrocytes  3.18   7.22 2.64e-21  1.14e-17
17 Setdb2 Oligodendrocytes  3.40   6.88 2.51e-21  1.14e-17
18 Gm3448 Oligodendrocytes  3.08   7.31 2.35e-21  1.14e-17
19 Gm3448              OPC  4.31   9.37 2.26e-43  3.75e-39
20 Ermard              OPC  3.46   9.87 3.14e-35  2.61e-31
21   Ccl2              OPC  8.33   7.51 9.83e-35  5.43e-31</code></pre>
</div>
<div id="pseudobulk-level-heatmap" class="section level1">
<h1>Pseudobulk-level heatmap</h1>
<pre class="r"><code># single gene across all clusters
pbHeatmap(sce, res, g = &quot;ENSMUSG00000035385.Ccl2&quot;)</code></pre>
<p><img src="figure/3-differential.Rmd/unnamed-chunk-2-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code># top-20 in single cluster
pbHeatmap(sce, res, k = &quot;Astrocytes&quot;)</code></pre>
<p><img src="figure/3-differential.Rmd/unnamed-chunk-2-2.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code># top-5 in ea. cluster
pbHeatmap(sce, res, top_n = 5)</code></pre>
<p><img src="figure/3-differential.Rmd/unnamed-chunk-2-3.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="violin-plots" class="section level1">
<h1>Violin plots</h1>
<pre class="r"><code># split cells by cluster
cs_by_k &lt;- split(colnames(sce), sce$cluster_id)
plotExpression(sce[, cs_by_k$Endothelial], 
    features = res_by_k$Endothelial$gene[seq_len(8)],
    x = &quot;sample_id&quot;, colour_by = &quot;group_id&quot;) + theme_classic() + 
    theme(axis.text.x = element_text(angle = 45, hjust = 1))</code></pre>
<p><img src="figure/3-differential.Rmd/unnamed-chunk-3-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="between-cluster-concordance" class="section level1">
<h1>Between-cluster concordance</h1>
<pre class="r"><code>ds_gs &lt;- lapply(res_by_k, pull, &quot;gene&quot;)
upset(fromList(ds_gs), sets = levels(sce$cluster_id))</code></pre>
<p><img src="figure/3-differential.Rmd/unnamed-chunk-4-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="write-results-to-.rds" class="section level1">
<h1>Write results to .rds</h1>
<pre class="r"><code>saveRDS(res, file.path(&quot;output&quot;, &quot;DS_results.rds&quot;))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.0 beta (2019-04-15 r76395)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] UpSetR_1.4.0                muscat_0.99.8              
 [3] scater_1.13.18              ggplot2_3.2.1              
 [5] SingleCellExperiment_1.7.7  SummarizedExperiment_1.15.8
 [7] DelayedArray_0.11.4         BiocParallel_1.19.2        
 [9] matrixStats_0.54.0          Biobase_2.45.0             
[11] GenomicRanges_1.37.14       GenomeInfoDb_1.21.1        
[13] IRanges_2.19.14             S4Vectors_0.23.20          
[15] BiocGenerics_0.31.5         dplyr_0.8.3                

loaded via a namespace (and not attached):
  [1] backports_1.1.4          circlize_0.4.7          
  [3] Hmisc_4.2-0              blme_1.0-4              
  [5] workflowr_1.4.0          plyr_1.8.4              
  [7] lazyeval_0.2.2           TMB_1.7.15              
  [9] splines_3.6.0            listenv_0.7.0           
 [11] digest_0.6.20            foreach_1.4.7           
 [13] htmltools_0.3.6          viridis_0.5.1           
 [15] gdata_2.18.0             lmerTest_3.1-0          
 [17] magrittr_1.5             checkmate_1.9.4         
 [19] memoise_1.1.0            cluster_2.1.0           
 [21] doParallel_1.0.15        limma_3.41.15           
 [23] globals_0.12.4           ComplexHeatmap_2.1.0    
 [25] annotate_1.63.0          prettyunits_1.0.2       
 [27] colorspace_1.4-1         blob_1.2.0              
 [29] xfun_0.9                 crayon_1.3.4            
 [31] RCurl_1.95-4.12          genefilter_1.67.1       
 [33] lme4_1.1-21              zeallot_0.1.0           
 [35] survival_2.44-1.1        iterators_1.0.12        
 [37] glue_1.3.1               gtable_0.3.0            
 [39] zlibbioc_1.31.0          XVector_0.25.0          
 [41] GetoptLong_0.1.7         BiocSingular_1.1.5      
 [43] future.apply_1.3.0       shape_1.4.4             
 [45] scales_1.0.0             DBI_1.0.0               
 [47] edgeR_3.27.13            Rcpp_1.0.2              
 [49] viridisLite_0.3.0        xtable_1.8-4            
 [51] progress_1.2.2           htmlTable_1.13.1        
 [53] clue_0.3-57              foreign_0.8-72          
 [55] bit_1.1-14               rsvd_1.0.2              
 [57] Formula_1.2-3            htmlwidgets_1.3         
 [59] gplots_3.0.1.1           RColorBrewer_1.1-2      
 [61] acepack_1.4.1            pkgconfig_2.0.2         
 [63] XML_3.98-1.20            nnet_7.3-12             
 [65] locfit_1.5-9.1           labeling_0.3            
 [67] tidyselect_0.2.5         rlang_0.4.0             
 [69] reshape2_1.4.3           AnnotationDbi_1.47.1    
 [71] munsell_0.5.0            tools_3.6.0             
 [73] RSQLite_2.1.2            evaluate_0.14           
 [75] stringr_1.4.0            yaml_2.2.0              
 [77] knitr_1.24               bit64_0.9-7             
 [79] fs_1.3.1                 caTools_1.17.1.2        
 [81] purrr_0.3.2              future_1.14.0           
 [83] nlme_3.1-141             pbkrtest_0.4-7          
 [85] compiler_3.6.0           rstudioapi_0.10         
 [87] beeswarm_0.2.3           png_0.1-7               
 [89] variancePartition_1.15.4 tibble_2.1.3            
 [91] geneplotter_1.63.0       stringi_1.4.3           
 [93] lattice_0.20-38          Matrix_1.2-17           
 [95] nloptr_1.2.1             vctrs_0.2.0             
 [97] pillar_1.4.2             GlobalOptions_0.1.0     
 [99] BiocNeighbors_1.3.3      cowplot_1.0.0           
[101] data.table_1.12.2        bitops_1.0-6            
[103] irlba_2.3.3              colorRamps_2.3          
[105] R6_2.4.0                 latticeExtra_0.6-28     
[107] KernSmooth_2.23-15       gridExtra_2.3           
[109] vipor_0.4.5              codetools_0.2-16        
[111] gtools_3.8.1             boot_1.3-23             
[113] MASS_7.3-51.4            assertthat_0.2.1        
[115] DESeq2_1.25.10           rprojroot_1.3-2         
[117] rjson_0.2.20             withr_2.1.2             
[119] sctransform_0.2.0        GenomeInfoDbData_1.2.1  
[121] hms_0.5.1                grid_3.6.0              
[123] rpart_4.1-15             glmmTMB_0.2.3           
[125] minqa_1.2.4              rmarkdown_1.15          
[127] DelayedMatrixStats_1.7.1 git2r_0.26.1            
[129] numDeriv_2016.8-1.1      base64enc_0.1-3         
[131] ggbeeswarm_0.6.0        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
